gfx.cpp

```diff
@@ -7262,7 +7262,7 @@ class GfxInternal
                             device_->CreateShaderResourceView(nullptr, &dummy_srv_desc, descriptors_.getCPUHandle(parameter.descriptor_slot_ + j));
                             continue;   // out of bounds mip level
                         }
-                        transitionResource(gfx_texture, D3D12_RESOURCE_STATE_PIXEL_SHADER_RESOURCE);
+                        transitionResource(gfx_texture, kernel.isGraphics() && kernel.ps_reflection_ != nullptr ? D3D12_RESOURCE_STATE_ALL_SHADER_RESOURCE : D3D12_RESOURCE_STATE_NON_PIXEL_SHADER_RESOURCE);
```

Comment:
This also fixes a bug in gfx whereby barriers arent functioning correctly when you have a compute shader called in a loop that ping-pongs inputs betwen `Texture2D` and `RWTexture2D` (such as a kernel theat creates mip maps with each kernel call setting the Texture2D input to equal the RWTexture2D output from the previous call). Previously this type of execution failed and required setting the input parameter as a `RWTexture2D` as a workaround